File: Php/app/User/Avatar/AvatarModelBuilder.php
<?php
namespace app\User\Avatar{

	use integration\Avatar\AvatarSize;
	use integration\Avatar\AvatarService;

	class AvatarModelBuilder {

		private $avatarService;

		public function __construct(AvatarService $avatarService){
			$this->avatarService = $avatarService;
		}

		public function build($email, $size = AvatarSize::large){
			$model = new AvatarModel();
			if($email != null){
				$model->avatarEnabled = true;
				$model->avatarUrl = $this->getAvatarUrl($email, $size);
			} else {
				$model->avatarEnabled = false;
			}
			return $model;
		}

		private function getAvatarUrl($email, $size){
			if($size == AvatarSize::small){
				return $this->avatarService->getSmallAvatarUrl($email);
			}
			return $this->avatarService->getLargeAvatarUrl($email);
		}

	}

}

File: Web/Models/PlayerModels/Details/AvatarModelBuilder.cs
using Core.Classes;
using Core.Services;

namespace Web.Models.PlayerModels.Details{

	class AvatarModelBuilder : IAvatarModelBuilder
	{
	    private readonly IAvatarService _avatarService;

	    public AvatarModelBuilder(IAvatarService avatarService)
		{
		    _avatarService = avatarService;
		}
        
	    public AvatarModel Build(string email, AvatarSize size = AvatarSize.Large){
			var model = new AvatarModel();
			if(email != null){
				model.AvatarEnabled = true;
                model.AvatarUrl = GetAvatarUrl(email, size);
			} else {
				model.AvatarEnabled = false;
			}
			return model;
		}

		private string GetAvatarUrl(string email, AvatarSize size){
			if(size == AvatarSize.Small){
				return _avatarService.getSmallAvatarUrl(email);
			}
			return _avatarService.getLargeAvatarUrl(email);
		}

	}

}